### ISSUE TITLE
Xcode 16 incompatibility

### ISSUE THREAD

**Issue by lgalfaso** [bazelbuild/bazel#23639]
### Description of the bug:

After updating to XCode 16, got the following error

ERROR: XXX/BUILD:7:11: Compiling xxx.cpp failed: absolute path inclusion(s) found in rule '//XXX':
the source file 'xxx.cpp' includes the following non-builtin files with absolute paths (if these are builtin files, make sure these paths are in your toolchain):
  '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/include/stdint.h'

### Which category does this issue belong to?

C++ Rules

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Update to Xcode 16, and compile anything that uses 'stdint.h' (I would expect that this issue is not limited to stdint.h)

### Which operating system are you running Bazel on?

MacOS 14.6.1

### What is the output of `bazel info release`?

release 7.3.1

**Comment by tjgq** [comment]
cc @keith

**Comment by fmeum** [comment]
Can you run `bazel fetch --configure` and see whether that fixes your build?

You may have to rerun this command if something non-hermetic (such as Xcode) changes on your machine.

**Comment by lgalfaso** [comment]
After running
```
% bazel fetch --configure   
INFO: All external dependencies fetched successfully.
Computing main repo mapping: 
```

I'm still seeing the same error.

**Comment by lgalfaso** [comment]
Running `bazel fetch --force --configure` did work :)

Thanks!

**Comment by fmeum** [comment]
@meteorcloudy The behavior of `bazel fetch --configure` doesn't match that of `bazel sync --configure`. Requiring the additional `--force` seems unnecessary, what do you think?

I also noticed that `--configure` seems to be ignored if the command has a residue. That seems confusing, how about we make this case an error?

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
